crates/ide/src/inlay_hints.rs

```diff
@@ -162,7 +162,7 @@ fn get_param_name_hints(
         .zip(args)
         .filter_map(|((param, _ty), arg)| {
             let param_name = match param? {
-                Either::Left(self_param) => self_param.to_string(),
+                Either::Left(self_param) => self_param.self_token()?.to_string(),
```

Comment:
Yeah, I think it's better to not include "&". But maybe just `"self".to_string()` then?

Reply:
This already excludes the reference stuff, since this is just the `self` identifier, or do you prefer not referencing the token at all?

Reply:
Sorry for being confusing! I mean that yes, current code is OK, but it could be simplified